src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -4742,9 +4794,75 @@ internal static object ThrowInvalidConversionException(object valueToConvert, Ty
                 return CacheConversion<object>(fromType, toType, LanguagePrimitives.ConvertIntegerToEnum, ConversionRank.Language);
             }
 
+            if (fromType == typeof(PSMethod) && toType.IsSubclassOf(typeof(MulticastDelegate)))
+            {
+                var psMethod = (PSMethod) fromInstance;
```

Comment:
As mentioned in long comment - I believe for this approach to work, you would **always** return something inside this `if`. Please do write a test to confirm what I've said, something like:

```PowerShell
class A {
    static [void] ggg([int] $i) {}
}
class B {
    static [void] fff([string] $i) {}
}

& { [Action[int]] [B]::fff } | Should Throw
& { [Action[int]] [A]::ggg } | Should Not Throw
```